### PR TITLE
Limit the number of images processed in alt_eval

### PR DESCRIPTION
**Pull request by sushen25**

### CODE REVIEW

File: alt_evaluator.py [hunk 1]
```diff
@@ -12,17 +13,28 @@ def __init__(self, path_to_lmdb_dir):
...
         with torch.no_grad():
 
             for i, (images, length_labels, digits_labels, paths) in enumerate(self._loader):
                 images, length_labels, digits_labels = images.cpu(), length_labels.cpu(), [digit_labels.cpu() for digit_labels in digits_labels]
                 length_logits, digit1_logits, digit2_logits, digit3_logits, digit4_logits, digit5_logits = model.eval()(images)
-                print("Evaluating images in batch: ", i+1)
+
+                batch_num = i+1
+                if batch_num > number_of_batches_to_evaluate:
+                    break
```
Comment (voirra) [inline alt_evaluator.py]: should this be higher so that it doesn't waste effort evaluating?

Comment (sushen25) [reply alt_evaluator.py]: number_of_batches_to_evaluated is determined by the number_of_images_to_evaluate:
```
number_of_batches_to_evaluate = math.inf
        if number_of_images_to_evaluate:
            number_of_batches_to_evaluate = math.ceil(int(number_of_images_to_evaluate) / self.batch_size)
            print("Number of batches to evaluate: ", number_of_batches_to_evaluate)
```

`number_of_images_to_evaluate` comes from the model academy api call in `tasks.py`:
```
params = {
        "dataset_path": base64.b64encode(b"dataset/data"),
    }
    if number_of_images_to_evaluate:
        params["number_of_images"] = number_of_images_to_evaluate
```

so the number of batches evaluated can be dynamically set by model academy, can be as many or as few as we like

File: alt_eval.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def _eval(path_to_checkpoint_file, path_to_data_dir, path_to_log_dir, lmdb_file,
     print(f'Evaluate {path_to_checkpoint_file} on {path_to_eval_lmdb_dir}')
     results = AltEvaluator(path_to_eval_lmdb_dir, number_of_images_to_evaluate).evaluate(model)
 
-    export_evaluate_to_data_dir(path_to_data_dir, results, get_model_version(path_to_checkpoint_file))
+    export_evaluate_to_data_dir(path_to_log_dir, results, get_model_version(path_to_checkpoint_file))
```
Comment (sushen25) [inline alt_eval.py]: The reason I'm storing the output in the datadir is so that I can access the json from model academy (we can change this once we've gotten rid of submodule.run(s) and pass the evaluate data through json) but for now I think just let the results be stored in the data dir instead of the log dir

tldr: we don't need this commit

File: alt_evaluator.py [hunk 3]
```diff
@@ -7,34 +7,27 @@
...
             transforms.CenterCrop([54, 54]),
             transforms.ToTensor(),
             transforms.Normalize([0.5, 0.5, 0.5], [0.5, 0.5, 0.5])
         ])
-        self.batch_size = 128
-        self._loader = torch.utils.data.DataLoader(Dataset(path_to_lmdb_dir, transform), batch_size=self.batch_size, shuffle=False)
+        self.dataset = Dataset(path_to_lmdb_dir, transform)
+        if number_images_to_evaluate:
+            self.dataset = self.dataset[0:int(number_images_to_evaluate)]
+        self._loader = torch.utils.data.DataLoader(self.dataset, batch_size=32, shuffle=False)
```
Comment (sushen25) [inline alt_evaluator.py]: Use `self.batch_size` here so we can dynamically change the batch size if we need

File: dataset.py [hunk 4]
```diff
@@ -19,20 +19,42 @@ def __len__(self):
         return self._length
 
     def __getitem__(self, index):
-        with self._reader.begin() as txn:
-            value = txn.get(self._keys[index])
+        if isinstance(index, slice):
```
Comment (sushen25) [inline dataset.py:22]: don't really understand this if statement, `__getitem__` should return only one data point (i.e. data at index) so why are you returning every single datapoint -> `result.append(image, length, digits, path)`?

Comment (voirra) [reply dataset.py:22]: `__getitem__` can also used to implement slice in this way

Comment (sushen25) [reply dataset.py:22]: so you can return the whole dataset in __getitem__?

Comment (voirra) [reply dataset.py:22]: It doesn't return the whole dataset, it will return based on indices of the `slice`, since slice has an `index.start` and `index.stop`

**Review by voirra — APPROVED** [review]
